src/theory/arith/theory_arith_private.cpp

```diff
@@ -2203,11 +2201,13 @@ bool TheoryArithPrivate::attemptSolveInteger(Theory::Effort effortLevel, bool em
 
   if(!options::trySolveIntStandardEffort()){ return false; }
 
-  if (d_lastContextIntegerAttempted <= (level >> 2)){
-
-    double d = (double)(d_solveIntMaybeHelp + 1) / (d_solveIntAttempts + 1 + level*level);
-    double t = fRand(0.0, 1.0);
-    if(t < d){
+  if (d_lastContextIntegerAttempted <= (level >> 2))
+  {
+    double d = (double)(d_solveIntMaybeHelp + 1)
+               / (d_solveIntAttempts + 1 + level * level);
+    double t = RNG::getRNG().pickDouble(0.0, 1.0);
+    if (t < d)
```

Comment:
Couldn't this be `pickWithProb`?

Reply:
Good question. Maybe @timothy-king can comment on this?

Reply:
pickWithProb seemse like it would work.